tests/test_domains/test_domain_py.py

```diff
@@ -508,6 +508,28 @@ def test_parse_annotation(app):
         ),
     )
 
+    doctree = _parse_annotation('*tuple[str, int]', app.env)
```

Comment:
Is `*tuple[...]` a valid stand-alone type annotation? If not, should we have some validation that `Starred` only comes after `[`?

I'm happy to keep things as currently proposed in this PR, just unsure which is the best approach.

Reply:
It is valid in the context of variadic generics annotations (see https://peps.python.org/pep-0646/#unpacking-tuple-types). In other places it is not, but I worked this out under the assumption that Sphinx is looking to accept all valid Python behaviors, not disallow non-valid ones (that would be neat, but would probably imply reimplementing a huge part of the parser in any Python implementation). 

With respect to what you are saying about validating it only comes after `[`: I came out with this PR to deal with annotations in type signatures in function arguments, not in type parameter specifications alla PEP 695. I am AFK and I don't remember off the top of my head if it does come into parsing the latter. I will look into it as soon as I can.

Reply:
Let's go with this for now, we can make changes later if need be. Thanks!

A